docs/ci-cd/generate_metrics.sh

```
@@ -0,0 +1,141 @@
+    
+    local count=$(echo "$runs" | jq -s 'length')
+    log "Fetched $count workflow runs"
+    echo "$runs"
+}
+
+fetch_commit_messages_batch() {
+    local runs_json=$1
+    
+    log "Fetching commit messages..."
+    # extract all unique SHAs and fetch commit messages
+    local unique_shas=$(echo "$runs_json" | jq -r '.[].sha' | sort -u)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)